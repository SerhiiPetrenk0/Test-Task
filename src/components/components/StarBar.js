import React from 'react';
import { StyledBsStarFill, StyledStars } from '../../styled/components/StarBar';
const colors = {
    orange: '#FFBA5A',
    grey: '#a9a9a9'
};

export const StarBar = props => {
    const stars = Array(5).fill(0)
    const { count } = props
    return (
        <StyledStars>
          {stars.map((_, index) => (
              <StyledBsStarFill
                key={index}
                size={20}
                color={count > index ? colors.orange : colors.grey}
              />
            ))}
        </StyledStars>
    );
}
