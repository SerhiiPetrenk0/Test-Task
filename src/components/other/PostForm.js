import React , { useState } from 'react';
import { Form , Button } from 'react-bootstrap';
import { postCommentAPI } from '../../api';
import { colors } from '../../styled/globalStyled';
import { 
    StyledContainer,
    StyledFormGroup,
    StyledBsStarFill
} from '../../styled/other/PostForm';

export const PostForm = () => {
    const [formValue, setFormValue] = useState(
        {
            id: 'B0721KZEEE',
            rating: '',
            coments: ''
        }
    );
    const [currentValue, setCurrentValue] = useState(0);
    const [hoverValue, setHoverValue] = useState(undefined);
    const stars = Array(5).fill(0);
  
    const handleClick = value => {
      setCurrentValue(value);
      setFormValue({...formValue, rating:value});
    };
  
    const handleMouseOver = newHoverValue => setHoverValue(newHoverValue);

    const handleMouseLeave = () => setHoverValue(undefined);

    const handleTextarea = val => setFormValue({...formValue, coments:val});

    const handleSubmit = event => {
        event.preventDefault();
        postCommentAPI(formValue);
    };

    return(
        <StyledContainer>
            <Form onSubmit={handleSubmit}>
                <StyledFormGroup controlId="exampleForm.ControlInput1">
                    {stars.map((_, index) => (
                            <StyledBsStarFill
                            key={index}
                            size={24}
                            onClick={() => handleClick(index + 1)}
                            onMouseOver={() => handleMouseOver(index + 1)}
                            onMouseLeave={handleMouseLeave}
                            color={(hoverValue || currentValue) > index ? colors.starActive : colors.starDisable}
                            />
                    ))}
                </StyledFormGroup>
                <StyledFormGroup controlId="exampleForm.ControlTextarea1">
                    <Form.Label>comment</Form.Label>
                    <Form.Control as="textarea" rows={3} value={formValue.coments} onChange={event => handleTextarea(event.target.value)} />
                </StyledFormGroup>
                  <Button variant="primary" type="submit">
                        Submit
                  </Button>
            </Form>
        </StyledContainer>
    );
};
