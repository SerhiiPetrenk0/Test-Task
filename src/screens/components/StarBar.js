import React from "react";
import { BsStarFill } from "react-icons/bs";

const colors = {
    orange: "#FFBA5A",
    grey: "#a9a9a9"
    
};

export const StarBar = (props) => {

    const stars = Array(5).fill(0)
    const { count } = props
    return (
        <div style={styles.stars}>
          {stars.map((_, index) => {
            return (
              <BsStarFill
                key={index}
                size={20}
                color={count > index ? colors.orange : colors.grey}
                style={{
                  marginRight: 5,
                  cursor: "pointer"
                }}
              />
            )
          })}
        </div>

    );
}

const styles = {
    stars: {
      display: "flex",
      flexDirection: "row",
    },
  };