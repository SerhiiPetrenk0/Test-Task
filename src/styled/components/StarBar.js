import styled from 'styled-components'

import { BsStarFill } from 'react-icons/bs';

export const StyledBsStarFill = styled(BsStarFill)`
    margin-right: 7px;
    cursor: pointer
`;
export const StyledStars = styled.div`
    display: flex;
    flexDirection: row,
`;
