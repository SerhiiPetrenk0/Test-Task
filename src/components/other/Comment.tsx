import React from 'react';
import { Card } from 'react-bootstrap';
import { StarBar } from './StarBar';
import { StyledCard } from '../../styled/other/Comment';

export const Comment = (props: { ListComment: any; }) => {
    const { ListComment } = props;
    return (
        <StyledCard>
            <Card.Body>
                <Card.Text>
                {ListComment.comments}
                </Card.Text>
                <Card.Subtitle>
                <StarBar count={ListComment.rating} />
                </Card.Subtitle>
            </Card.Body>
        </StyledCard>
    );
};
