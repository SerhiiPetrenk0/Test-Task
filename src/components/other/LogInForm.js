import React from "react"
import * as Yup from 'yup'
import { Form, Button, Modal } from "react-bootstrap"
import { useDispatch  } from "react-redux"
import { useFormik } from "formik"
import { StyledValidEmail } from "../../styled/other/LogInForm"
import { fillUser } from "../../redux/redusers/userInfoReduser"

export const LogInForm = props => {
    const { show, handleClose } = props
    const dispatch = useDispatch()
    const formik = useFormik({
      initialValues:{
        email: '',
        password: ''
      },
      validationSchema: Yup.object({
        email: Yup
        .string()
        .max(30, 'Max simbol 15))')
        .required('* this field is required')
        .matches(/^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/ , "Invalid email"),
        password: Yup
        .string()
        .required('passwordrequired')
        .matches(/^(?=.*[0-9])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{6,16}$/ , 'FALSE PASSWORD')
      }),
      onSubmit: values => {
        console.log(values)
        dispatch(fillUser(formik.values))
        handleClose()
      }
    })
    //почистити код / повиносити змінні / проблеми із змінною check / регулярки в змінні / 
      const emaillog = !!formik.errors.email ? formik.errors.email : "We'll never share your email with anyone else."
      const passwordlog = !!formik.errors.password ? formik.errors.password : "Your password must be 6 to 16 characters long, contain letters, numbers, special characters !@#$%^&*, And no spaces or emojis."
      const emailMassage = <StyledValidEmail check={!!formik.errors.email}>{emaillog}</StyledValidEmail>
      const passwordMassage = <StyledValidEmail check={!!formik.errors.password}>{passwordlog}</StyledValidEmail>
     console.log(!!formik.errors.password)
    return (
<>
  <Modal show={show} onHide={handleClose} backdrop="static" keyboard={false}>
    <Modal.Header closeButton>
      <Modal.Title>Modal title</Modal.Title>
    </Modal.Header>
    <Modal.Body>
      <Form onSubmit={formik.handleSubmit}>
        <Form.Group className="mb-3" controlId="formBasicEmail">
          <Form.Label>Email address</Form.Label>
          <Form.Control name="email" onChange={formik.handleChange} value={formik.values.email} type="email" placeholder="Enter email" />
          {emailMassage}
        </Form.Group>

        <Form.Group className="mb-3" controlId="formBasicPassword">
          <Form.Label>Password</Form.Label>
          <Form.Control name="password" onChange={formik.handleChange} value={formik.values.password} type="password" placeholder="Password" />
          {passwordMassage}
        </Form.Group>

        <Button variant="primary" type="submit">
          Submit
        </Button>
      </Form>
    </Modal.Body>
  </Modal>
</>
    )
} 