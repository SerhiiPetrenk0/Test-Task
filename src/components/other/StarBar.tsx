import React from 'react';

import { StyledBsStarFill, StyledStars } from '../../styled/other/StarBar';
import { colors } from '../../styled/globalStyled';

interface TypeProps {
  count: string
}

export const StarBar: React.FC<TypeProps> = props => {
    const stars: number[] = Array(5).fill(0);
    const { count } = props;
    return (
        <StyledStars>
          {stars.map((_: number, index: number) => (
              <StyledBsStarFill
                key={index}
                size={20}
                color={parseInt(count, 10) > index ? colors.starActive : colors.starDisable}
              />
            ))}
        </StyledStars>
    );
};
