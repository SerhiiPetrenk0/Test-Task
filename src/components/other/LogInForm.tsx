import React from 'react';
import { Form, Button, Modal } from 'react-bootstrap';
import { useDispatch  } from 'react-redux';
import { useFormik } from 'formik';

import { StyledValidEmail } from '../../styled/other/LogInForm';
import { fillUser } from '../../redux/ducks/userInfoDuck';
import { TypeFormikLogInForm, TypeUserinfo } from '../../interface';
import { YopLoginForm } from '../../validation';

interface TypeProps {
  show: boolean,
  handleClose: () => void
}

export const LogInForm: React.FC<TypeProps> = props => {
    const { show, handleClose } = props;
    const dispatch = useDispatch();
    const formik: TypeFormikLogInForm = useFormik({
      initialValues: {
        email: '',
        password: ''
      },
      validationSchema: YopLoginForm,
      onSubmit: (values: TypeUserinfo) => {
        dispatch(fillUser(formik.values))
        handleClose()
      }
    });
    
    const emailLog = !!formik.errors.email ?
      formik.errors.email :
      "We'll never share your email with anyone else.";
    const emailMassage: JSX.Element = (
      <StyledValidEmail check={formik.errors.email}>
        {emailLog}
      </StyledValidEmail>
    );

    const passwordLog = !!formik.errors.password ?
      formik.errors.password : 
      "Your password must be 6 to 16 characters long, contain letters, numbers, special characters !@#$%^&*, And no spaces or emojis."
    const passwordMassage: JSX.Element = (
      <StyledValidEmail check={formik.errors.password}>
        {passwordLog}
      </StyledValidEmail>
    );

    return (
      <>
        <Modal 
          show={show} 
          onHide={handleClose} 
          backdrop="static" 
          keyboard={false}
        >
          <Modal.Header closeButton>
            <Modal.Title>Registration</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <Form onSubmit={formik.handleSubmit}>
              <Form.Group className="mb-3" controlId="formBasicEmail">
                <Form.Label>Email address</Form.Label>
                <Form.Control 
                  name="email" 
                  onChange={formik.handleChange} 
                  value={formik.values.email} 
                  type="email" 
                  placeholder="Enter email" 
                />
                {emailMassage}
              </Form.Group>

              <Form.Group className="mb-3" controlId="formBasicPassword">
                <Form.Label>Password</Form.Label>
                <Form.Control 
                  name="password" 
                  onChange={formik.handleChange} 
                  value={formik.values.password} 
                  type="password" 
                  placeholder="Password" 
                />
                {passwordMassage}
              </Form.Group>

              <Button variant="primary" type="submit">
                Submit
              </Button>
            </Form>
          </Modal.Body>
        </Modal>
      </>
    );
} ;
