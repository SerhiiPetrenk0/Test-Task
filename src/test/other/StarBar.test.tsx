import React from "react";
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import { StarBar } from '../../components/other/StarBar';

configure({ adapter: new Adapter() })

describe('StarBar component', () => {
  it('Snapshot StarBar component count={"0"}', () => {
    const component = shallow(<StarBar count={"0"} />)
    expect(toJson(component)).toMatchSnapshot();
  })
  it('Snapshot StarBar component count={"4"}', () => {
    const component = shallow(<StarBar count={"4"} />)
    expect(toJson(component)).toMatchSnapshot();
  })
});