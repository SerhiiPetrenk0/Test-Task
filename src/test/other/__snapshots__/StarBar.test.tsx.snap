// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarBar component Snapshot StarBar component count={"0"} 1`] = `
<styled.div>
  <Styled(BsStarFill)
    color="#a9a9a9"
    key="0"
    size={20}
  />
  <Styled(BsStarFill)
    color="#a9a9a9"
    key="1"
    size={20}
  />
  <Styled(BsStarFill)
    color="#a9a9a9"
    key="2"
    size={20}
  />
  <Styled(BsStarFill)
    color="#a9a9a9"
    key="3"
    size={20}
  />
  <Styled(BsStarFill)
    color="#a9a9a9"
    key="4"
    size={20}
  />
</styled.div>
`;

exports[`StarBar component Snapshot StarBar component count={"4"} 1`] = `
<styled.div>
  <Styled(BsStarFill)
    color="#FFBA5A"
    key="0"
    size={20}
  />
  <Styled(BsStarFill)
    color="#FFBA5A"
    key="1"
    size={20}
  />
  <Styled(BsStarFill)
    color="#FFBA5A"
    key="2"
    size={20}
  />
  <Styled(BsStarFill)
    color="#FFBA5A"
    key="3"
    size={20}
  />
  <Styled(BsStarFill)
    color="#a9a9a9"
    key="4"
    size={20}
  />
</styled.div>
`;
